fili-core/src/main/java/com/yahoo/bard/webservice/logging/blocks/DruidFilterInfo.java

```diff
@@ -0,0 +1,66 @@
+// Copyright 2016 Yahoo Inc.
+// Licensed under the terms of the Apache license. Please see LICENSE.md file distributed with this work for terms.
+package com.yahoo.bard.webservice.logging.blocks;
+
+import com.yahoo.bard.webservice.druid.model.filter.MultiClauseFilter;
+import com.yahoo.bard.webservice.druid.model.filter.NotFilter;
+import com.yahoo.bard.webservice.logging.LogInfo;
+
+import com.fasterxml.jackson.annotation.JsonAutoDetect;
+
+import com.yahoo.bard.webservice.druid.model.filter.Filter;
+
+import java.util.ArrayDeque;
+import java.util.Collections;
+import java.util.Deque;
+import java.util.LinkedHashMap;
+import java.util.Map;
+
+/**
+ * Logs some structural data about the filter sent to Druid, without actually logging the entire (potentially massive)
+ * filter.
+ */
+@JsonAutoDetect(fieldVisibility = JsonAutoDetect.Visibility.ANY)
+public class DruidFilterInfo implements LogInfo {
+
+    protected final Map<String, Long> numEachFilterType;
+
+    /**
+     * Constructor.
+     *
+     * @param filter  The filter that needs to be analyzed
+     */
+    public DruidFilterInfo(Filter filter) {
+        numEachFilterType = buildFilterCount(filter);
+    }
+
+    /**
+     * Performs a DFS search of the filter tree, populating the specified map with the number of instances of each
+     * filter type appearing in the filter.
+     *
+     * @param filter  The filter that needs to be traversed
+     *
+     * @return A map containing a count of each type of filter
+     */
+    private Map<String, Long> buildFilterCount(Filter filter) {
+        if (filter == null) {
+            return Collections.emptyMap();
+        }
+        Map<String, Long> filterTypeCounter = new LinkedHashMap<>();
+        Deque<Filter> filterStack = new ArrayDeque<>();
+        filterStack.add(filter);
+        while (!filterStack.isEmpty()) {
+            Filter currentFilter = filterStack.pop();
+            filterTypeCounter.put(
+                    currentFilter.getClass().getSimpleName(),
+                    filterTypeCounter.getOrDefault(currentFilter.getClass().getSimpleName(), 0L) + 1L
+            );
+            if (currentFilter instanceof MultiClauseFilter) {
+                filterStack.addAll(((MultiClauseFilter) currentFilter).getFields());
+            } else if (currentFilter instanceof NotFilter) {
+                filterStack.add(((NotFilter) currentFilter).getField());
+            }
```

Comment:
This smells a little like extensibility pain... It would be great if Java made it easy to ask if things had various methods, but about the best we can do is add interfaces for specific methods. To that end, do we want to make `HasFields` and `HasField` interfaces, apply those interfaces to the `MultiVlauseFilter` and `NotFilter` (respectively), and then update this conditional to depend on those interfaces instead of these more concrete classes?

Specifically, it's the `NotFilter` dependency that drew my attention, since the `MultiClauseFilter` is abstract already and (roughly) intended to serve the same purpose as the interface would.